ush/cam/cam_util.py

```diff
@@ -319,19 +320,66 @@ def format_filler(unfilled_file_format, valid_time_dt, init_time_dt,
                                           filled_file_format_chunk)
     return filled_file_format
 
-def get_completed_jobs(completed_jobs_file):
-    completed_jobs = set()
-    if os.path.exists(completed_jobs_file):
-        with open(completed_jobs_file, 'r') as f:
-            completed_jobs = set(f.read().splitlines())
+def get_completed_jobs(completed_jobs_dir, job_type=''):
+    """
+    Returns a set of job names (e.g., 'job1', 'job2') that are marked complete
+    in completed_jobs_dir.
+    """
+    if not job_type:
+        job_dir = Path(completed_jobs_dir)
+    else:
+        job_dir = Path(completed_jobs_dir) / job_type
+
+    if job_dir.is_dir():
+        completed_jobs = {
+            p.name for p in job_dir.iterdir() 
+            if p.is_file() and p.name.startswith("job")
+        }
+    else:
+        completed_jobs = {}
+
     return completed_jobs
 
-def mark_job_completed(completed_jobs_file, job_name, job_type=""):
-    with open(completed_jobs_file, 'a') as f:
+def mark_job_completed(restart_dir, data_dir, verif_case, 
+                       completed_jobs_dirname, job_name, job_type=''):
+    """
+    Marks a job as completed by creating a blank file at:
+        data_dir/verif_case/completed_jobs_dirname[/job_type]/job_name
+    If SENDCOM is set, copies the file to:
+        restart_dir/completed_jobs_dirname[/job_type]/job_name
+    """ 
+
+    SENDCOM = os.environ.get('SENDCOM')
+    if SENDCOM is None:
+        e = f"FATAL ERROR: SENDCOM is not defined in the job card for {job_name}"
         if job_type:
-            f.write(job_type + "_" + job_name + "\n")
-        else:
-            f.write(job_name + "\n")
+            e+=f" (type: {job_type})"
+        raise ValueError(e)
+
+    restart_out = Path(restart_dir) / completed_jobs_dirname
+    data_out = Path(data_dir) / verif_case
+    if job_type:
+        restart_out = restart_out / job_type
+        data_out = data_out / 'METplus_output' / 'workdirs' / job_type / job_name / completed_jobs_dirname / job_type
+    else:
+        data_out = data_out / 'out' / 'workdirs' / job_name / completed_jobs_dirname
+
+    if not data_out.is_dir():
+        e = f"FATAL ERROR: Completed jobs directory does not exist: {data_out}"
+        raise FileNotFoundError(e)
+
+    job_file = data_out / job_name
+
+    # Create an empty file to mark completion
+    job_file.touch(exist_ok=True)
+
```

Comment:
@MarcelCaron-NOAA @AliciaBentley-NOAA I've been looking at the code in this PR to see if I could use it for mesoscale, and I noticed the empty completed_jobs files.  I do not think that empty files are allowed by NCO, so I think that we'd need to have something inside this file - like the job name or the word "complete" or something.  I just wanted to note that.